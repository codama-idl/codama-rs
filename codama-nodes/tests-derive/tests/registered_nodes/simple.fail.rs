use codama_nodes::{HasKind, NodeTrait, NodeUnionTrait};
use codama_nodes_derive::{node, node_union, RegisteredNodes};

#[node]
pub struct NumberTypeNode {}

#[node]
pub struct StructTypeNode {}

#[node]
pub struct StructFieldTypeNode {}

#[derive(RegisteredNodes)]
#[node_union]
pub enum RegisteredTypeNode {
    Number(NumberTypeNode),
    Struct(StructTypeNode),
    #[registered]
    StructField(StructFieldTypeNode),
}

fn main() {
    TypeNode::StructField(StructFieldTypeNode {});
}
