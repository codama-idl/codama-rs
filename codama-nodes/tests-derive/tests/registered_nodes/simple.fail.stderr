error[E0599]: no variant or associated item named `StructField` found for enum `TypeNode` in the current scope
  --> tests/registered_nodes/simple.fail.rs:23:15
   |
15 | pub enum RegisteredTypeNode {
   | --------------------------- variant or associated item `StructField` not found for this enum
...
23 |     TypeNode::StructField(StructFieldTypeNode {});
   |               ^^^^^^^^^^^ variant or associated item not found in `TypeNode`
